### DISCUSSION TITLE
pluginlib API to implement image processing features

### DISCUSSION THREAD

**Discussion by twdragon** [ros-drivers/usb_cam#226] [Ideas]
Should the node provide a `pluginlib`-based API and mechanism to perform frame-based postprocessing of the image before being published? Please share your opinions here!

**Comment by twdragon** [comment]
@knorth55 @flynneva in the meantime, could you please share what you think about this?

**Comment by flynneva** [comment]
@twdragon for me this sounds like scope creep and something that violates the single responsibility principle.

This packages purpose is purely meant to grab images from v4l devices and publish it to ROS, that's it.

**Comment by knorth55** [comment]
im not sure about your idea but is it like image_transport plugin?
if so, it is better to make another package with image_transport plugin

**Comment by twdragon** [comment]
@flynneva @knorth55 the idea is to provide lightweight API to make some basic processing, like flipping for example, avoiding republishing and using the `image_proc` package

**Comment by knorth55** [comment]
@twdragon i think it should be separate package. this package is just for grabbing image from usb device and publish.
if you want to do more postprocessing, you should make another node or nodelet in another package.

**Comment by twdragon** [comment]
OK, here we should keep the singular responsibility, so I close this discussion
